src/extension/wps-download/src/main/java/org/geoserver/wps/gs/download/RasterEstimator.java

```diff
@@ -250,4 +258,24 @@ public boolean execute(
         }
         return true;
     }
+
+    private void reloadBands(CoverageInfo ci) throws Exception {
+        String nativeName = ci.getNativeCoverageName();
+        CatalogBuilder cb = new CatalogBuilder(catalog);
+        cb.setStore(ci.getStore());
+        MetadataMap metadata = ci.getMetadata();
+        CoverageInfo rebuilt;
+        if (metadata != null && metadata.containsKey(CoverageView.COVERAGE_VIEW)) {
+            GridCoverage2DReader reader =
+                    (GridCoverage2DReader)
+                            catalog.getResourcePool()
+                                    .getGridCoverageReader(
+                                            ci, nativeName, GeoTools.getDefaultHints());
+            rebuilt = cb.buildCoverage(reader, nativeName, null);
+        } else {
+            rebuilt = cb.buildCoverage(nativeName);
+        }
+        ci.getDimensions().clear();
```

Comment:
This update is not really saving any new information in the catalog, and it's not intended to either (having config that gets modified as a result of a WPS process meant to download stuff would be unexpected).

Maybe it's best to just return rebuilt.getDimensions() and assign that to the dimension list instead.